bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,16 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r"""
+            (https?://                     # http:// or https://
+            [^.:%\s_/?#[\]@\\]+            # Initial Part of Domain
+            \.                             # A required dot (.)
+            (?:[^\s()%\'"`<>|\\]+          # Domain / Subdomain Name
+            |                              # OR
+            \([^\s()%\'"`<>|\\]*\))*)      # Subdomain Name contained within ()
+            (?<!\.)(?<!,)                  # URL shouldn't end with (.) or (,)
```

Comment:
I think this should be: Exclude trailing dot or comma from the URL.

Or does it reject URLSLs ending in dot or comma?